### PR TITLE
Increase freelist.releaseRange unit test coverage.

### PR DESCRIPTION
**Pull request by jpbetz**
Adds a test table covering some of the boundary conditions for `freelist.releaseRange`.

cc @heyitsanthony

### CODE REVIEW

**Comment by jpbetz** [comment]
cc @mml, @cheftako

File: freelist_test.go [hunk 1]
```diff
@@ -44,6 +44,131 @@ func TestFreelist_release(t *testing.T) {
...
+			testPage{id: 5, n: 1, allocTxn: 125, freeTxn: 150},
+			testPage{id: 6, n: 1, allocTxn: 150, freeTxn: 175},
+			testPage{id: 7, n: 1, allocTxn: 175, freeTxn: 200},
+		},
+		releaseRanges: []testRange{testRange{50, 149}, testRange{151, 300}},
+		wantFree:      []pgid{4, 7},
+	},
+}
+
+// Ensure that releaseRange handles boundry conditions correctly
```
Comment (heyitsanthony) [inline freelist_test.go]: s/boundry/boundary

File: freelist_test.go [hunk 2]
```diff
@@ -44,6 +44,131 @@ func TestFreelist_release(t *testing.T) {
 	}
 }
 
+type testRange struct {
```
Comment (xiang90) [inline freelist_test.go]: why do you define these types out of the func body? do you plan to reuse them in other tests?

File: freelist_test.go [hunk 3]
```diff
@@ -44,6 +44,131 @@ func TestFreelist_release(t *testing.T) {
...
+	},
+	{
+		title: "Adjacent pending and read transactions at 199, 200",
+		pagesIn: []testPage{
+			testPage{id: 3, n: 1, allocTxn: 199, freeTxn: 200},
+			testPage{id: 4, n: 1, allocTxn: 200, freeTxn: 201},
+		},
+		releaseRanges: []testRange{
+			testRange{100, 198},
+			testRange{200, 199},
```
Comment (cheftako) [inline freelist_test.go]: Is this range supposed to be out of order? In fact should this range even be here even correctly ordered?

File: freelist_test.go (hunk 2 above)
Comment (jpbetz) [reply freelist_test.go]: I'll move them into the test function. Thanks!

File: freelist_test.go (hunk 3 above)
Comment (jpbetz) [reply freelist_test.go]: I'll add a comment. This is intended to simulate the ranges `db.freePages` might produce.

File: freelist_test.go [hunk 4]
```diff
@@ -44,6 +44,131 @@ func TestFreelist_release(t *testing.T) {
...
+		wantFree: nil,
+	},
+	{
+		title: "Multiple pending, read transaction at 150",
+		pagesIn: []testPage{
+			testPage{id: 3, n: 1, allocTxn: 100, freeTxn: 200},
+			testPage{id: 4, n: 1, allocTxn: 100, freeTxn: 125},
+			testPage{id: 5, n: 1, allocTxn: 125, freeTxn: 150},
+			testPage{id: 6, n: 1, allocTxn: 150, freeTxn: 175},
+			testPage{id: 7, n: 1, allocTxn: 175, freeTxn: 200},
```
Comment (cheftako) [inline freelist_test.go]: Can we have a testPage where the allocTxn is in the first release range and the freeTxn is in the second releaseRange? (Eg 125, 175)

Comment (jpbetz) [reply freelist_test.go]: Excellent. Added.

**Comment by xiang90** [comment]
lgtm
